src/lem/eval.rs

```diff
@@ -1160,22 +1160,29 @@ fn apply_cont(cprocs: &[(&Symbol, usize)], ivc: bool) -> Func {
     func!(apply_cont(result, env, cont, ctrl): 4 => {
         match symbol ctrl {
             "apply-continuation" => {
-                let ret = Symbol("return");
-                let err: Cont::Error;
-                let term: Cont::Terminal;
-                let cont_is_term = eq_tag(cont, term);
-                let cont_is_err = eq_tag(cont, err);
-                let cont_is_term_or_err = or(cont_is_term, cont_is_err);
-                if cont_is_term_or_err {
-                    return (result, env, cont, ret)
-                }
```

Comment:
This is unnecessary because `reduce` already sets the control directly to `ret` in case of termination or error